packages/ag-charts-community/src/scale/ordinalTimeScale.ts

```diff
@@ -212,6 +213,13 @@ export class OrdinalTimeScale extends BandScale<Date, TimeInterval | number> {
             d = new Date(d);
         }
 
+        if (typeof d === 'string') {
+            const maybeDate = new Date(d);
+            if (isValidDate(maybeDate)) {
+                d = maybeDate;
+            }
+        }
```

Comment:
When a `Date` is put through `JSON.stringify()` it converts to a string. When `Annotations.convertPoint()` then tries to convert it using the scale, it needs to be turned back into a date (if valid).

Alternatively, the memento could carry the type information in addition to the value, and make the restoration of types be the originators responsibility?

Reply:
I'm thinking we should either:
- Include the original type in the persisted state
  - maybe even a convention of a `__date` suffix on the property name would be sufficient (stripped out on restore)?
  - use a more complex `object` structure for values with a `type`.
- Alternatively, keep a scheme of the expected property types for use during save/restore, which we can validate against and use to work out deserialised type?

Reply:
I've updated it to use the replacer and receiver functions on `JSON.stringify()` and `JSON.parse()`. It transforms `Date` objects into `{ __type: 'date', value: ... }` and back to dates.